src/Ocelot/Requester/Middleware/HttpRequesterMiddleware.cs

```diff
@@ -12,20 +13,33 @@ public class HttpRequesterMiddleware : OcelotMiddleware
         private readonly RequestDelegate _next;
         private readonly IHttpRequester _requester;
         private readonly IOcelotLogger _logger;
+        private readonly DiagnosticSource _diagnostics;
 
         public HttpRequesterMiddleware(RequestDelegate next,
             IOcelotLoggerFactory loggerFactory,
             IHttpRequester requester, 
-            IRequestScopedDataRepository requestScopedDataRepository)
+            IRequestScopedDataRepository requestScopedDataRepository,
+            DiagnosticSource diagnostic)
             :base(requestScopedDataRepository)
         {
             _next = next;
             _requester = requester;
+            _diagnostics = diagnostic;
             _logger = loggerFactory.CreateLogger<HttpRequesterMiddleware>();
         }
 
         public async Task Invoke(HttpContext context)
         {
+            if (_diagnostics.IsEnabled("Ocelot.Responder.Middleware.HttpRequesterMiddlewareStarting"))

```

Comment:
Can we move this code into the DiagnosticListener?

Something like...

```csharp
  [DiagnosticName("Microsoft.AspNetCore.MiddlewareAnalysis.MiddlewareStarting")]
        public virtual void OnMiddlewareStarting(HttpContext httpContext, string name)
        {
            //not sure about if statement but this gets called when each middleware starts.
            _source.Write($"{name}MiddlewareStarting",
                new
                {
                    httpContext = httpContext
                }); 
        }
```

Reply:
yes， move this cde into OcelotDiagnosticListener